api/v1alpha1/clientprofilemapping_types.go

```diff
@@ -0,0 +1,76 @@
+/*
+Copyright 2024.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// BlockPoolRefSpec indentify a block pool - client profile pair
+type BlockPoolRefSpec struct {
+	//+kubebuilder:validation:Required
+	ClientProfileName string `json:"clientProfileName,omitempty"`
+
+	//+kubebuilder:validation:Required
+	//+kubebuilder:validation:Minimum:=0
+	PoolId int `json:"poolId,omitempty"`
+}
+
+// BlockPoolMappingSpec define a mapiing between a local and remote
+// block pools
+type BlockPoolMappingSpec struct {
+	//+kubebuilder:validation:Required
+	Local BlockPoolRefSpec `json:"local,omitempty"`
+
+	//+kubebuilder:validation:Required
+	Remote BlockPoolRefSpec `json:"remote,omitempty"`
+}
+
+// ClientProfileMappingSpec defines the desired state of ClientProfileMapping
+type ClientProfileMappingSpec struct {
+	//+kubebuilder:validation:Optional
```

Comment:
any specific reason to make it optional?

Reply:
Yes. The profile mapping spec was designed for future extensibility for adding additional types of mapping. If we make the block pool mapping mandatory we will need to change our contract when we introduce a additional mapping sections